drivers/sx127x/sx127x_getset.c

```diff
@@ -0,0 +1,735 @@
+        }
+        break;
+    }
+
+    return air_time;
+}
+
+void sx127x_set_sleep(sx127x_t *dev)
+{
+    /* Disable running timers */
+    xtimer_remove(&dev->_internal.tx_timeout_timer);
+    xtimer_remove(&dev->_internal.rx_timeout_timer);
+
+    /* Put chip into sleep */
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_SLEEP);
+    sx127x_set_state(dev,  SX127X_RF_IDLE);
+}
+
+void sx127x_set_standby(sx127x_t *dev)
+{
+    /* Disable running timers */
+    xtimer_remove(&dev->_internal.tx_timeout_timer);
+    xtimer_remove(&dev->_internal.rx_timeout_timer);
+
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_STANDBY);
+    sx127x_set_state(dev,  SX127X_RF_IDLE);
+}
+
+void sx127x_set_rx(sx127x_t *dev)
+{
+    switch (dev->settings.modem) {
+        case SX127X_MODEM_FSK:
+            /* FSK not supported yet */
+            return;
+
+        case SX127X_MODEM_LORA:
+        {
+            if (dev->settings.lora.iq_inverted) {
+                sx127x_reg_write(dev, SX127X_REG_LR_INVERTIQ,
+                                 ((sx127x_reg_read(dev, SX127X_REG_LR_INVERTIQ)
+                                   & SX127X_RF_LORA_INVERTIQ_TX_MASK & SX127X_RF_LORA_INVERTIQ_RX_MASK)
+                                  | SX127X_RF_LORA_INVERTIQ_RX_ON | SX127X_RF_LORA_INVERTIQ_TX_OFF));
+                sx127x_reg_write(dev, SX127X_REG_LR_INVERTIQ2, SX127X_RF_LORA_INVERTIQ2_ON);
+            }
+            else {
+                sx127x_reg_write(dev,
+                                 SX127X_REG_LR_INVERTIQ,
+                                 ((sx127x_reg_read(dev, SX127X_REG_LR_INVERTIQ)
+                                   & SX127X_RF_LORA_INVERTIQ_TX_MASK & SX127X_RF_LORA_INVERTIQ_RX_MASK)
+                                  | SX127X_RF_LORA_INVERTIQ_RX_OFF | SX127X_RF_LORA_INVERTIQ_TX_OFF));
+                sx127x_reg_write(dev, SX127X_REG_LR_INVERTIQ2, SX127X_RF_LORA_INVERTIQ2_OFF);
+            }
+
+            /* ERRATA 2.3 - Receiver Spurious Reception of a LoRa Signal */
+            if (dev->settings.lora.bandwidth < 9) {
+                sx127x_reg_write(dev, SX127X_REG_LR_DETECTOPTIMIZE,
+                                 sx127x_reg_read(dev, SX127X_REG_LR_DETECTOPTIMIZE) & 0x7F);
+                sx127x_reg_write(dev, SX127X_REG_LR_TEST30, 0x00);
+                switch (dev->settings.lora.bandwidth) {
+                    case SX127X_BW_125_KHZ: /* 125 kHz */
+                        sx127x_reg_write(dev, SX127X_REG_LR_TEST2F, 0x40);
+                        break;
+                    case SX127X_BW_250_KHZ: /* 250 kHz */
+                        sx127x_reg_write(dev, SX127X_REG_LR_TEST2F, 0x40);
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+            else {
+                sx127x_reg_write(dev, SX127X_REG_LR_DETECTOPTIMIZE,
+                                 sx127x_reg_read(dev, SX127X_REG_LR_DETECTOPTIMIZE) | 0x80);
```

Comment:
This is black magic ! This bit is not documented in both SX1272 and SX1276.
But Semtech set this bit (for SX1276 only)
What it is this bit ?